java/system-test/system-test-dsl/src/test/java/sleeper/systemtest/dsl/testutil/InMemoryTestInstance.java

```diff
@@ -76,6 +77,7 @@ private static SystemTestInstanceConfiguration usingSystemTestDefaultsWithProper
     public static InstanceProperties createDslInstanceProperties() {
         InstanceProperties instanceProperties = createTestInstanceProperties();
         instanceProperties.set(RETAIN_INFRA_AFTER_DESTROY, "false");
+        instanceProperties.set(RETAIN_LOGS_AFTER_DESTROY, "false");
```

Comment:
To clarify, this also needs changing to address my previous comment: https://github.com/gchq/sleeper/pull/5965#discussion_r2494825373

> We need to keep the logs after a system test instance is destroyed, otherwise we wouldn't be able to see what happened in the nightly tests.

It looks like the assertion for this has been removed from the unit test now as well. I think that doesn't seem quite right, as SetupInstanceTest, SetupInstanceIT and SetupInstanceST should all match.